lib/wapm-contract/src/parser.rs

```diff
@@ -0,0 +1,380 @@
+    fn parse_fn_export() {
+        let parse_res = fn_export("(fn \"name\" (param f64 i32) (result f64 i32))").unwrap();
+        assert_eq!(
+            parse_res,
+            (
+                "",
+                Export::Fn {
+                    name: "name".to_string(),
+                    params: vec![WasmType::F64, WasmType::I32],
+                    result: vec![WasmType::F64, WasmType::I32],
+                }
+            )
+        );
+    }
+
+    #[test]
+    fn parse_imports_test() {
+        let parse_res =
+            parse_imports("(assert-import (fn \"ns\" \"name\" (param f64 i32) (result f64 i32)))")
+                .unwrap();
+        assert_eq!(
+            parse_res,
+            (
+                "",
+                vec![Import::Fn {
+                    namespace: "ns".to_string(),
+                    name: "name".to_string(),
+                    params: vec![WasmType::F64, WasmType::I32],
+                    result: vec![WasmType::F64, WasmType::I32],
+                }]
+            )
+        );
+
+        let parse_res = parse_imports(
+            "(assert-import (fn \"ns\" \"name\"  
+                                               (param f64 i32) (result f64 i32))
+    ( global \"length\" ( type 
+i32 )
+)
+                                          (fn \"ns\" \"name2\" (param f32
+                                                                      i64)
+                                                                (
+                                                                 result
+                                                                 f64
+                                                                 i32
+                                                                 )
+                                          ) 
+)",
+        )
+        .unwrap();
+        assert_eq!(
+            parse_res,
+            (
+                "",
+                vec![
+                    Import::Fn {
+                        namespace: "ns".to_string(),
+                        name: "name".to_string(),
+                        params: vec![WasmType::F64, WasmType::I32],
+                        result: vec![WasmType::F64, WasmType::I32],
+                    },
+                    Import::Global {
+                        name: "length".to_string(),
+                        var_type: WasmType::I32,
+                    },
+                    Import::Fn {
+                        namespace: "ns".to_string(),
+                        name: "name2".to_string(),
+                        params: vec![WasmType::F32, WasmType::I64],
+                        result: vec![WasmType::F64, WasmType::I32],
+                    },
+                ]
+            )
+        );
+    }
+
+    #[test]
+    fn top_level_test() {
+        let parse_res = parse_contract(
+            " (assert-import (fn \"ns\" \"name\" (param f64 i32) (result f64 i32)))
+ (assert-export (fn \"name2\" (param) (result i32)))
```

Comment:
To comply with thee WebAssembly test spec, I think it will be better `assert_import` and `assert_export` (using `_` instead of `-`)

Reply:
Also use `func` instead of `fn`